hazelcast/src/main/java/com/hazelcast/map/impl/operation/AddIndexOperation.java

```diff
@@ -49,6 +51,26 @@ public AddIndexOperation(String name, String attributeName, boolean ordered) {
         this.ordered = ordered;
     }
 
+    @Override
+    public boolean shouldBackup() {
+        return mapContainer.getTotalBackupCount() > 0;
+    }
+
+    @Override
+    public int getSyncBackupCount() {
+        return mapContainer.getTotalBackupCount();
```

Comment:
is there a reason to report the total backup count here instead of `mapContainer.getBackupCount()`? Same applies for using `mapContainer.getAsyncBackupCount()` instead of `0` below?

Reply:
Not sure, the idea was to make all backup partitions aware of the index being created when the operation finishes. `AddIndexOperation` is not that frequent and most likely nobody expects high level of performance from it.

Reply:
Agreed, it seems safer to get the index information on backup partitions synchronously. It shouldn't make much difference with a stable cluster, while with an unstable cluster a sync backup failure would alert the user.